src/core/sal/gem5/Gem5Controller.cc

```diff
@@ -36,7 +36,7 @@ Gem5Controller::~Gem5Controller()
 		delete *it;
 		it = m_CPUs.erase(it);
 	}
-	delete m_Mem;
+	delete &getMemoryManager();
```

Comment:
At this point only the MemoryManager for MEMTYPE_RAM would be deleted, since this is the default for getMemoryManager().